module/zcommon/zpool_prop.c

```diff
@@ -304,6 +304,21 @@ vdev_prop_init(void)
 		{ NULL }
 	};
 
+	static const zprop_index_t vdev_failfast_table[] = {
+		{ "none",		VDEV_FAILFAST_NONE },
+		{ "device",		VDEV_FAILFAST_DEVICE },
+		{ "device,driver",
+		    VDEV_FAILFAST_DEVICE | VDEV_FAILFAST_DRIVER },
+		{ "device,transport",
+		    VDEV_FAILFAST_DEVICE | VDEV_FAILFAST_TRANSPORT },
+		{ "driver",		VDEV_FAILFAST_DRIVER },
+		{ "driver,transport",
+		    VDEV_FAILFAST_DRIVER | VDEV_FAILFAST_TRANSPORT },
+		{ "transport",		VDEV_FAILFAST_TRANSPORT },
+		{ "all",		VDEV_FAILFAST_ALL },
+		{ NULL }
+	};
```

Comment:
From user perspective I honestly can't imagine how would I use all those combinations.

Reply:
Yes, based on the conversation we had at the dev summit:

we'll switch the property to just on/off. This will also make it more portable to FreeBSD, if we add a similar flag to avoid retries at the lower levels.

We'll add a module parameter to allow the user to override the default mask so they can set these combinations if they REALLY need, but it doesn't clutter up the interface for most users.